src/client/js/components/circle-chart.js

```diff
@@ -0,0 +1,223 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+const CHART_RADIUS = 50
+const CHART_DIAMETER = CHART_RADIUS * 2
+const CHART_CIRCUMFERENCE = Math.PI * CHART_DIAMETER
+const CHART_UPDATE_DURATION = 250
+
+const styles = `
+<style>
+  .circle-chart {
+    align-items: center;
+    display: flex;
+    gap: var(--padding-md);
+    margin: 0;
+    transition: opacity ${CHART_UPDATE_DURATION * 0.5}ms ease;
+  }
+
+  .circle-chart.updating {
+    opacity: 0
+  }
+
+  .circle-chart-title {
+    display: block;
+    font-family: Inter, Inter-fallback, sans-serif;
+    margin-bottom: var(--padding-xs);
+  }
+
+  .circle-chart-label {
+    align-items: center;
+    color: var(--gray-40);
+    display: flex;
+    font-size: 0.875rem;
+    gap: var(--padding-sm);
+    position: relative;
+  }
+
+  .circle-chart-label::before {
+    color: inherit;
+    content: '\\2B24'; /* Black Large Circle */
+    font-size: 0.65em;
+    padding-bottom: 0.175em;
+  }
+
+  .circle-chart svg {
+    border-radius: 50%;
+    height: 10vw;
+    min-height: 100px;
+    min-width: 100px;
+    transform: rotate(-90deg);
+    width: 10vw;
+  }
+
+  .circle-chart circle {
+    cx: 50%;
+    cy: 50%;
+    fill: none;
+    r: 50%;
+  }
+
+  .circle-chart text {
+    font-family: metropolis, sans-serif;
+    font-weight: 700;
+    text-anchor: middle;
+    transform: rotate(90deg);
+    transform-origin: center;
+  }
+</style>
+`
+
+const calcPercentage = (total, value) => {
+  if (!total) {
+    return 0
+  }
+
+  return parseFloat((value / total).toFixed(3, 10))
+}
+
+const html = () => `
+  ${styles}
+  <figure class='circle-chart'></figure>
+`
+
+customElements.define('circle-chart', class extends HTMLElement {
+  static get observedAttributes () { return ['data-chart'] }
```

Comment:
Only take the attribute `data-chart` which contains all of the necessary info to render the charts for our current use cases. Happy to discuss a declarative and more intuitive API, but with regard to our time constraints I’d like to do this when we actually had the chance to use them on the actual pages. Would this approach work for you @toufali?

Reply:
Updated the attributes that the custom component takes as discussed.